provider/scripts/offline_store_spark_runner.py

```diff
@@ -23,20 +33,67 @@ def execute_sql_query(job_type, output_uri, sql_query, source_list):
             output_uri_with_timestamp = f'{output_uri}{dt}'
 
             output_dataframe.coalesce(1).write.option("header", "true").mode("overwrite").parquet(output_uri_with_timestamp)
-    except Exception as ident:
-        print(ident)
+            return output_uri_with_timestamp
+    except Exception as e:
+        print(e)
+        raise e
+
+def execute_df_job(output_uri, code, sources):
+    spark = SparkSession.builder.appName("Dataframe Transformation").getOrCreate()
+    
+    func_parameters = {}
+    for name, location in sources.items():
+        func_parameters[name] = spark.read.parquet(location)
     
+    try:
+        # code = dill.loads(bytes(code))
+        func = types.FunctionType(code, globals(), "df_transformation")
```

Comment:
What does this look like? Is it the entire transformation function or just the dataframe it returns?

Reply:
it is the entire transformation function. Here is the [test case](https://github.com/featureform/featureform/blob/feature/spark-df-transformations-pyscript/provider/scripts/tests/conftest.py#L71-L75).